### PR TITLE
fix #2 (bad type of some parameters for some z80/68k low-level commands)

### LINKED ISSUE #2: libticables slv_bulk_read timeout
Sort of getting desperate here.

I've tried to get tilp to work on OSX, Win and Linux today, and still haven't succeeded. The ultimate goal is to install KnightOS.
On OSX and on Linux I managed to get tilp to connect to the calculator (TI 84+, boot code 1.02; OS 2.55MP), but as soon as I try to dump the ROM (both via double click on operating system and via Tools->dump Rom), I get a slv_bulk_read (Operation timed out) error.
Any ideas?

This has happened on OSX Mavericks, and ArchLinux, with tilp 1.16 and 1.17 and the newest libticables from the AUR.

### PR DESCRIPTION
**Pull request by adriweb**
And also add some safety flags to the cmake build...

### CODE REVIEW

File: libticalcs/trunk/src/cmdz80.cc [hunk 1]
```diff
@@ -540,7 +540,10 @@ int ti82_send_asm_exec(CalcHandle* handle, VarEntry * var)
 	endptr = (var->name[0] == 0x24 ? fpBase : tempMem);
 	buffer[8] = 0x2a; buffer[9] = LSB(endptr); buffer[10] = MSB(endptr);
 	/* ld de, -program_size */
+#pragma warning(push)
```
Comment (debrouxl) [inline libticalcs/trunk/src/cmdz80.cc:543]: `#pragma` isn't portable in the first place, but would `#pragma msvc ...` be possible at least for the `warning(disable:4146)` ?

Comment (adriweb) [reply libticalcs/trunk/src/cmdz80.cc:543]: Well I know, but gcc and clang just warn that it's an unknown pragma and ignore it anyway :P

But contrary to those two compilers with a compiler-name-as-first-arg, it looks like `#pragma msvc` isn't a thing...
So I suppose we could add `#ifdef _MSC_VER` etc. but that's a total of 5 extra lines just to disable a warning, bleh.

Comment (debrouxl) [reply libticalcs/trunk/src/cmdz80.cc:543]: Taking a step back... fixing the code would be better :)
offset should be `int32_t`, and after adding an early bounds check against < 2 and >= 32768 (both models have 32 KB of physical RAM) on `var->size` below `VALIDATE_VARENTRY(var)`, something like `offset = -(int32_t)((uint16_t)var->size - 2);` should do the job, hopefully without the compilers complaining about numerical value truncation or overflows.

Comment (adriweb) [reply libticalcs/trunk/src/cmdz80.cc:543]: Yeah that's probably better. We'll have to compare the actual output in `buffer` just in case...

**Comment by adriweb** [comment]
I'll close this PR since e5ad813ccf8bae71de6aa344df020178698b57b3 was integrated.
But we still need to do something about the ASM generation part...

**Pull request closed without merging**